pyproject.toml

```diff
@@ -12,46 +12,44 @@ readme = "README.md"
 license = {file = "LICENSE"}
 classifiers = [
     "Programming Language :: Python :: 3",
-    "Programming Language :: Python :: 3.9",
     "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
     "License :: OSI Approved :: MIT License",
 ]
 keywords = ["OTE", "OTEAPI", "LITE", "OTEAPI-DLITE"]
-requires-python = ">=3.9"
+requires-python = "~=3.10"
 dynamic = ["version"]
 
 dependencies = [
     # Core dependencies
-    "DLite-Python>=0.4.5,<0.5.24",
-    "numpy>=1.21,<3",
-    "oteapi-core>=0.7.0.dev3,<1",
-    "pydantic>=2.1,<3",
-    "pydantic-settings>=2.0,<3",
-    "typing-extensions>=4.8,<5; python_version<'3.10'",
+    "DLite-Python>=0.5.29",
+    "numpy~=2.1",
+    "oteapi-core~=1.0",
+    "pydantic-settings~=2.8",
     "Pillow>=9.0.1,<12",
-    "tripper>=0.3.4, <0.3.5",
+    "tripper[mappings]>=0.4.0",
```

Comment:
I prefer having an upper limits too, which is then checked with dependabot. Much easier to identify when things go wrong...